src/ahocorasick/AhoCorasickPlus.cpp

```diff
@@ -72,10 +77,15 @@ AhoCorasickPlus::EnumReturnStatus AhoCorasickPlus::addPattern
 }
 
 AhoCorasickPlus::EnumReturnStatus AhoCorasickPlus::addPattern
-    (const char pattern[], PatternId id)
+    (const std::string &pattern, PatternId id, bool copy)
```

Comment:
If the function above became a `string_view`, these two functions could just go away I think.